php/class-wp-seo-post-meta-boxes.php

```diff
@@ -0,0 +1,235 @@
+	protected function setup() {
+		add_action( 'add_meta_boxes', array( $this, 'add_meta_boxes' ), 10, 2 );
+		add_action( 'save_post', array( $this, 'save_post' ), 10, 3 );
+		add_action( 'edit_attachment', array( $this, 'edit_attachment' ) );
+		add_action( 'add_attachment', array( $this, 'add_attachment' ) );
+	}
+
+	/**
+	 * Fires after all built-in meta boxes have been added to an edit screen.
+	 *
+	 * @param string $post_type Post type.
+	 * @param WP_Post $post Post object.
+	 */
+	public function add_meta_boxes( $post_type, $post ) {
+		$this->register_meta_box( $post_type );
+	}
+
+	/**
+	 * Post meta box callback.
+	 *
+	 * @param WP_Post $post The post being edited.
+	 */
+	public function do_meta_box( $post ) {
+		$this->render_meta_box( $post );
+	}
+
+	/**
+	 * Fires once a post has been saved.
+	 *
+	 * @param int $post_ID Post ID.
+	 * @param WP_Post $post Post object.
+	 * @param bool $update Whether this is an existing post being updated.
+	 */
+	public function save_post( $post_ID, $post, $update ) {
+		$this->save_fields( $post_ID );
+	}
+
+	/**
+	 * Fires once an attachment has been added.
+	 *
+	 * @param int $post_ID Attachment ID.
+	 */
+	public function add_attachment( $post_ID ) {
+		$this->save_fields( $post_ID );
+	}
+
+	/**
+	 * Fires once an existing attachment has been updated.
+	 *
+	 * @param int $post_ID Attachment ID.
+	 */
+	public function edit_attachment( $post_ID ) {
+		$this->save_fields( $post_ID );
+	}
+
+	/**
+	 * Register a WP SEO post meta box.
+	 *
+	 * @param string $post_type Post type to register the meta box with.
+	 */
+	private function register_meta_box( $post_type ) {
+		if ( ! WP_SEO_Settings()->has_post_fields( $post_type ) ) {
+			return;
+		}
+
+		add_meta_box(
+			'wp_seo',
+			wp_seo_meta_box_heading(),
+			array( $this, 'do_meta_box' ),
+			$post_type,
+			/**
+			 * Filter the screen context where the fields should display.
+			 *
+			 * @param string @see add_meta_box().
+			 */
+			apply_filters( 'wp_seo_meta_box_context', 'normal' ),
+			/**
+			 * Filter the display priority of the fields within the context.
+			 *
+			 * @param string @see add_meta_box().
+			 */
+			apply_filters( 'wp_seo_meta_box_priority', 'high' )
+		);
+	}
+
+	/**
+	 * Display the SEO fields for a post.
+	 *
+	 * @param WP_Post $post The post being edited.
+	 */
+	private function render_meta_box( $post ) {
+		$title = get_post_meta( $post->ID, '_meta_title', true );
+		$description = get_post_meta( $post->ID, '_meta_description', true );
+		wp_nonce_field( plugin_basename( __FILE__ ), 'wp-seo-nonce' );
+		?>
+			<table class="wp-seo-post-meta-fields">
+				<tbody>
+					<tr>
+						<th scope="row"><label for="wp_seo_meta_title"><?php esc_html_e( 'Title Tag', 'wp-seo' ); ?></label></th>
+						<td>
+							<input type="text" id="wp_seo_meta_title" name="seo_meta[title]" value="<?php echo esc_attr( $title ); ?>" size="96" />
+							<div>
+								<?php esc_html_e( 'Title character count: ', 'wp-seo' ); ?>
```

Comment:
I would remove the trailing whitespace in this localized string (line 157 too). If we depend on it, it would be too easy for a translator to forget it.